### PR TITLE
AI response should be in embed

### LINKED ISSUE #917: ChatGPT Auto-answer should be in an embed
Its necessary for better UX that ChatGPT answers are posted in embeds with a proper author and icon that clearly marks the content as AI answer.

Right now, its too confusing for users since it looks like a regular message.

The disadvantage is that code snippets are typically harder to read in embeds. We will try it out and see how bad it is. Its likely that we can live with it.

Also, when posting embeds, the text limit is boosted from 2k to 4k characters (? check embed max size). So that would reduce the amount of messages that need to be posted, which is good.

### LINKED ISSUE #928: ChatGpt Auto-Answer posts multiple mini messages
Sometimes, ChatGPTs messages are split into multiple very short messages that are then send individually. As if it wouldnt split by pure message length, but by indicators like newline or similar triggers.

This hits rate limits and also does not play well with the Dismiss button.

We have to investigate the code and the splitting logic to find the source of the issue.

### PR DESCRIPTION
**Pull request by ankitsmt211**
* helper method to generate embed
* remove AI parser class that partitioned responses & related tests
* refactor dismiss button on generated responses
* update docs
* resolves #917 and possibly #935 (since now response embed contains title)

* resolves #928 

Here's how new AI generated responses look like: 

gpt-response-embed

### CODE REVIEW

**Review by SquidXTV — APPROVED** [review]
Looks good to me :D
Btw you want to change the user icon to chatgpt icon?
Are we even allowed to?

**Comment by ankitsmt211** [comment]
>  Btw you want to change the user icon to chatgpt icon? Are we even allowed to?

that would be really nice if we could do that, i can't think of a way to do that. Prolly would need store random asset somewhere and get a url perhaps that we can pass for author.

**Comment by SquidXTV** [comment]
> > Btw you want to change the user icon to chatgpt icon? Are we even allowed to?
> 
> that would be really nice if we could do that, i can't think of a way to do that. Prolly would need store random asset somewhere and get a url perhaps that we can pass for author.

https://en.m.wikipedia.org/wiki/File:ChatGPT_logo.svg
https://chat.openai.com/apple-touch-icon.png
https://chat.openai.com/favicon-32x32.png

**Comment by ankitsmt211** [comment]
I will give these a try, also i linked a new issue #928 since it will be resolved as well so we can touch 3 of the issues with this merge.

File: application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java [hunk 1]
```diff
@@ -143,27 +143,43 @@ RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
...
+
+        int embedTitleLimit = MessageEmbed.TITLE_MAX_LENGTH;
+        String capitalizedTitle = Character.toUpperCase(title.charAt(0)) + title.substring(1);
+
+        String titleForEmbed = capitalizedTitle.length() > embedTitleLimit
+                ? capitalizedTitle.substring(0, embedTitleLimit)
+                : capitalizedTitle;
+
+        return new EmbedBuilder()
+            .setAuthor(selfUser.getName(), null, selfUser.getEffectiveAvatarUrl())
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java:173]: maybe add gpt as author?

**Review by Taz03 — CHANGES_REQUESTED** [review]

**Review by Taz03 — APPROVED** [review]

**Pull request merged**